src/sentry/static/sentry/app/views/settings/projectIssueGrouping/upgradeGrouping.tsx

```diff
@@ -0,0 +1,145 @@
+
+import {getGroupingChanges, getGroupingRisk} from './utils';
+
+type Props = {
+  groupingConfigs: EventGroupingConfig[];
+  groupingEnhancementBases: GroupingEnhancementBase[];
+  organization: Organization;
+  projectId: string;
+  project: Project;
+  onUpgrade: () => void;
+  api: Client;
+};
+
+function UpgradeGrouping({
+  groupingConfigs,
+  groupingEnhancementBases,
+  organization,
+  projectId,
+  project,
+  onUpgrade,
+  api,
+}: Props) {
+  const hasAccess = organization.access.includes('project:write');
+  const {
+    updateNotes,
+    riskLevel,
+    latestGroupingConfig,
+    latestEnhancementsBase,
+  } = getGroupingChanges(project, groupingConfigs, groupingEnhancementBases);
+  const {riskNote, alertType} = getGroupingRisk(riskLevel);
+  const noUpdates = !latestGroupingConfig && !latestEnhancementsBase;
+
+  const newData: Record<string, string> = {};
+  if (latestGroupingConfig) {
+    newData.groupingConfig = latestGroupingConfig.id;
+  }
+  if (latestEnhancementsBase) {
+    newData.groupingEnhancementsBase = latestEnhancementsBase.id;
+  }
+
+  const handleUpgrade = () => {
+    addLoadingMessage(t('Changing grouping\u2026'));
+    api
+      .requestPromise(`/projects/${organization.slug}/${projectId}/`, {
+        method: 'PUT',
+        data: newData,
+      })
+      .then(resp => {
+        clearIndicators();
+        ProjectActions.updateSuccess(resp);
+        onUpgrade();
+      }, handleXhrErrorResponse(t('Unable to upgrade config')));
+  };
+
+  if (!groupingConfigs || !groupingEnhancementBases) {
+    return null;
+  }
+
+  function getModalMessage() {
+    return (
+      <React.Fragment>
+        <TextBlock>
+          <strong>{t('Upgrade Grouping Strategy')}</strong>
+        </TextBlock>
+        <TextBlock>
+          {t(
+            'You can upgrade the grouping strategy to the latest but this is an irreversible operation.'
+          )}
+        </TextBlock>
+        <TextBlock>
+          <strong>{t('New Behavior')}</strong>
+          <div dangerouslySetInnerHTML={{__html: marked(updateNotes)}} />
+        </TextBlock>
+        <TextBlock>
+          <Alert type={alertType}>{riskNote}</Alert>
+        </TextBlock>
+      </React.Fragment>
+    );
+  }
+
+  return (
+    <Panel id="upgrade-grouping">
+      <PanelHeader>{t('Upgrade Grouping')}</PanelHeader>
+      <PanelBody>
+        <Field
+          label={t('Upgrade Grouping Strategy')}
+          help={tct(
+            'If the project uses an old grouping strategy an update is possible.[linebreak]Doing so will cause new events to group differently.',
+            {
+              linebreak: <br />,
+            }
+          )}
+          disabled
+        >
+          <Confirm
+            disabled={noUpdates}
+            onConfirm={handleUpgrade}
+            priority={riskLevel >= 2 ? 'danger' : 'primary'}
+            confirmText={t('Upgrade')}
+            message={getModalMessage()}
+          >
+            <div>
```

Comment:
do you really need this div here?

Reply:
yeah, tooltips are otherwise not positioned correctly